lib/include/ert/nexus/nexus_plot.hpp

```diff
@@ -39,25 +44,270 @@ struct unexpected_eof : public std::runtime_error {
     using std::runtime_error::runtime_error;
 };
 
-class NexusPlot {
-public:
-    NexusPlot( const std::string& );
-    NexusPlot( std::istream& );
-
-    int32_t num_classes = 0;
+struct NexusHeader {
+    int32_t num_classes;
     int32_t day, month, year;
     int32_t nx, ny, nz;
     int32_t ncomp;
-    std::vector<std::string> class_names;
-    std::vector<int32_t> vars_in_class;
-    std::vector< std::vector<std::string> > var_names;
+};
+
 
-    ecl_sum_type* ecl_summary( const std::string& ecl_case );
+struct NexusData {
+    int32_t timestep;
+    float time;
+    int32_t max_perfs;
+    std::array< char, 8 > classname;
+    std::array< char, 8 > instancename;
+    std::array< char, 4 > varname;
+    float value;
 
-private:
-    void load( std::istream& );
+    constexpr bool operator==(const NexusData& rhs) const noexcept {
+        return this->timestep      == rhs.timestep &&
+               this->time          == rhs.time &&
+               this->max_perfs     == rhs.max_perfs &&
+               this->classname     == rhs.classname &&
+               this->instancename == rhs.instancename &&
+               this->varname       == rhs.varname &&
+               this->value         == rhs.value;
+    }
+    constexpr bool operator!=(const NexusData& rhs) const noexcept {
+        return ! (*this == rhs);
+    }
+    // bool operator< (const NexusData& rhs) { }
+    // bool operator<=(const NexusData& rhs) { }
+    // bool operator> (const NexusData& rhs) { }
+    // bool operator>=(const NexusData& rhs) { }
 };
 
+namespace get {
+
+    std::array< char, 8 > classname( const NexusData& nd ) {
+        return nd.classname;
+    }
+    std::array< char, 8 > instancename( const NexusData& nd ) {
```

Comment:
The `NexusData` is already a struct with all members public; I think the attributes should just be accessed through `.xxx` notatiation.

Reply:
These are used for filtering and transforming iterables, they are just convenience functions. I use them several places.

Reply:
I agree it seems a bit much, but it does make the rest of the code much less messy.